### PR TITLE
docs: update README.md for v3

### PR DESCRIPTION
**Pull request by KazuCocoa**
Closes https://github.com/appium/python-client/issues/911

- https://github.com/appium/python-client/pull/909

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +58,55 @@ For example, some changes in the Selenium binding could break the Appium client.
 > We strongly recommend you manage dependencies with version management tools such as Pipenv and requirements.txt
 > to keep compatible version combinations.
 
+
+### Quick migration guide from v2 to v3
+- Replacement
+    - `start_activity`: Please use [`mobile: startActivity`](https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-startactivity)
+    - `launch_app`, `close_app`, `reset`: Please refer to https://github.com/appium/appium/issues/15807
+    - `available_ime_engines`, `is_ime_active`, `activate_ime_engine`, `deactivate_ime_engine`, `active_ime_engine`: Please use [`mobile: shell`](https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-shell)
+    - `set_value`, `set_text`: Please use `send_keys` or W3C actions' `send_keys`
```
Comment (mykola-mokhnach) [inline README.md:67]: extra quite here?

File: README.md [hunk 2]
```diff
@@ -101,6 +58,55 @@ For example, some changes in the Selenium binding could break the Appium client.
...
+- Removal
+    - `end_test_coverage` is no longer available
+    - `session` is no longer available
+    - `all_sessions` is no longer available
+
+### Quick migration guide from v1 to v2
```
Comment (mykola-mokhnach) [inline README.md:73]: Perhaps we could just create a link in the readme but move migration details themselves into a separate doc

File: README.md [hunk 3]
```diff
@@ -101,6 +58,55 @@ For example, some changes in the Selenium binding could break the Appium client.
...
+
+### Quick migration guide from v2 to v3
+- Replacement
+    - `start_activity`: Please use [`mobile: startActivity`](https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-startactivity)
+    - `launch_app`, `close_app`, `reset`: Please refer to https://github.com/appium/appium/issues/15807
+    - `available_ime_engines`, `is_ime_active`, `activate_ime_engine`, `deactivate_ime_engine`, `active_ime_engine`: Please use [`mobile: shell`](https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-shell)
+    - `set_value`, `set_text`: Please use `send_keys` or W3C actions' `send_keys`
+- Removal
+    - `end_test_coverage` is no longer available
+    - `session` is no longer available
```
Comment (mykola-mokhnach) [inline README.md:70]: end_test_coverage was a methods and two below were properties

File: README.md (hunk 2 above)
Comment (KazuCocoa) [reply README.md:73]: Sounds good, let me create a separate doc as a followup pr

**Pull request merged**